appcues/src/main/java/com/appcues/data/AppcuesRepository.kt

```diff
@@ -208,4 +212,34 @@ internal class AppcuesRepository(
             post(queue, current)
         }
     }
+
+    private fun mapExperience(
```

Comment:
this function was moved out from L161 to it's own function, since it got larger and having all this inline would cause detekt issues as well.

the key aspect is the try/catch now, where it handles AppcuesTraitException during mapping, and returns back the failed experience so that it can properly track the error - the actual error message is inserted on L236 from the exception message